Lib/distutils/ccompiler.py

```diff
@@ -341,8 +341,7 @@ def _setup_compile(self, outdir, macros, incdirs, sources, depends,
         pp_opts = gen_preprocess_options(macros, incdirs)
 
         build = {}
-        for i in range(len(sources)):
-            src = sources[i]
+        for i, src in enumerate(sources):
```

Comment:
This stell leaves the `objects`, why not just `for src, obj in zip(sources, objects):`

Reply:
I tried to stay close to the original code and do what was needed to replace ```range(len(sources))``` with ```enumerate(sources)```.
Your suggestion could be done but it looks like the comments in the bug tracker would prefer no changes at all.